Common/Thread/ParallelLoop.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <functional>
+#include <mutex>
+#include <condition_variable>
+
+#include "Common/Thread/ThreadManager.h"
+
+// Same as the latch from C++21, just counting upwards for no particular reason.
+struct WaitableCounter : public Waitable {
+public:
+	WaitableCounter(int count) : count_(count) {}
+
+	void Count() {
+		std::unique_lock<std::mutex> lock(mutex_);
+		if (count_ == 0) {
+			return;
+		}
+		count_--;
+		if (count_ == 0) {
+			// We were the last one to increment
+			cond_.notify_all();
+		}
+	}
+
+	void Wait() override {
+		std::unique_lock<std::mutex> lock(mutex_);
+		if (count_ == 0) {
+			return;
+		}
+		cond_.wait(lock);
```

Comment:
Should this be a while in case of a spurious wake?

-[Unknown]

Reply:
Yeah, it should. Doing that.